Utils/MessageUtils.cs

```diff
@@ -1,119 +1,182 @@
+                var sourceMessage = gossipLines[i].Split(';');
+                var destinationMessage = gossipLines[i + 1].Split(';');
+                var nextGossip = new Gossip
+                {
+                    SourceMessage = sourceMessage,
+                    DestinationMessage = destinationMessage
                 };
-                messageTable.Add(textId, message);
-                baseAddress += 8;
-            }
 
-            return messageTable;
+                gossipList.Add(nextGossip);
+            }
+            return gossipList;
         }
 
-        private static bool IsBadMessage(string message)
-        {
-            return message.Contains("a segment of health") || message.Contains("currency") ||
-                message.Contains("money") || message.Contains("cash") ||
-                message.Contains("wealth") || message.Contains("riches and stuff") ||
-                message.Contains("increased life");
-        }
+        public static List<MessageEntry> MakeGossipQuotes(Settings settings, List<ItemObject> items, Random random)
+        {
+            if (!settings.EnableGossipHints)
+                return new List<MessageEntry>();
+
+            var hints = new List<string>();
+            var GossipList = GetGossipList();
 
-        public static void WriteGossipHints(List<string> hints, Random random)
-        {
-            var messageList = GetMessageTable();
-            for (ushort textId = GOSSIP_START_ID; textId < GOSSIP_END_ID; textId++)
+            foreach (var item in items)
             {
-                if (GossipExclude.Contains(textId)
-                    || !messageList.TryGetValue(textId, out MessageEntry message))
+                if (!item.ReplacesAnotherItem)
                 {
                     continue;
                 }
 
-                int selectedIndex;
-                string selectedHint;
-                int length = message.Size + 1;
-                do
+                // Skip hints for vanilla bottle content
+                if ((!settings.RandomizeBottleCatchContents)
+                    && ItemUtils.IsBottleCatchContent(item.ID))
                 {
-                    selectedIndex = random.Next(hints.Count);
-                    selectedHint = hints[selectedIndex];
+                    continue;
+                }
 
-                    if (IsBadMessage(selectedHint) && random.Next(8) != 0)
-                    {
-                        continue;
-                    }
+                // Skip hints for vanilla shop items
+                if ((!settings.AddShopItems)
+                    && ItemUtils.IsShopItem(item.ID))
+                {
+                    continue;
+                }
 
-                    length = selectedHint.Length + MessageHeader.Count;
+                // Skip hints for vanilla dungeon items
+                if (!settings.AddDungeonItems
+                    && ItemUtils.IsDungeonItem(item.ID))
```

Comment:
Unrelated from your changes - I look forward to the point where we no longer rely on the item ints, we could change all `ItemUtils` functions to being extensions on the itemobjects.